pages/help/HelpHandler.inc.php

```diff
@@ -64,15 +82,45 @@ function index($args, $request) {
 		$parser->url_filter_func = function ($url) use ($filename) {
 			return dirname($filename) . '/' . $url;
 		};
+
+		if (!$isPluginHelpRequest) {
+			$content = file_get_contents($path . $filename);
+
+			// If the toc is requested, add the additional help chapters
+			if (preg_match('#../SUMMARY\.md#', $filename)) {
+				$content = rtrim($content);
+				$content .= "\n## Plugins\n";
+
+				foreach (array_keys($pluginHelpChapters) as $key) {
+					$content .= "   * [$key]($pluginHelpChapters[$key])";
+				}
+			}
+		} else {
+			$content = file_get_contents($this->getPathForPluginHelpChapter($request, $language));
+		}
+
 		return new JSONMessage(
 			true,
 			array(
-				'content' => $parser->transform(file_get_contents($path . $filename)),
+				'content' => $parser->transform($content),
 				'previous' => $previousLink,
 				'next' => $nextLink,
 			)
 		);
 	}
+
+	/**
+	 * Return the path of the plugin help chapter
+	 * @param $request PKPRequest
+	 * @return string Path of the help chapter
+	 */
+	function getPathForPluginHelpChapter($request, $language) {
+		$language = AppLocale::getIso1FromLocale(AppLocale::getLocale());
+		$reqArgs = $request->getRequestedArgs();
+		$pathParts = array_slice($reqArgs, 1);
+		array_splice($pathParts, count($pathParts) - 1, 0, $language);
+		return join("/", $pathParts);
+	}
```

Comment:
Cool. I'd like to be more forceful about how we expect help documents to be structured and linked to. So right now it's set up to do something like this:

```
{help file="plugins/generic/citationStyleLanguage/help/citation-style-language.md"}
```

Let's add some specific parameters here for plugins, so that there is less confusion around languages and we always force plugin developers to use a `/help` directory:

```
{help plugin="generic/citationStyleLanguage" file="citation-style-language.md"}
```

And as a bonus, let's implicitly recognize an `index.md` file, so the following will automatically look into `<plugin>/help/<language>/index.md`:

```
{help plugin="generic/citationStyleLanguage"}
```

Doing it this way will force plugin developers to follow a consistent pattern for where help documents are stored, and make clear our expectations around multi-lingual support. This might also mean you have to do less string manipulation to handle the languages.

If you're not familiar with it, you'll find the function that renders `{help}` in `PKPTemplateManager::smartyHelp`.